compiler/testData/codegen/box/javaInterop/notNullAssertions/enhancedNullability/inLambdaReturnWithExpectedType.kt

```diff
@@ -1,4 +1,5 @@
 // !LANGUAGE: +StrictJavaNullabilityAssertions
+// IGNORE_BACKEND_FIR: JVM_IR
```

Comment:
I'll take a look at this regression. This fails at line 36, which seems correct to me in the sense that a function that said `@NotNull` actually returns `null`. But I guess this is related to this change: https://github.com/JetBrains/kotlin/commit/ee020ef2903ca25154973ff7d9f93a1213068bf2

That is, if the expected type has `@EnhancedNullability` (in this case, I guess so), according to that change, we shouldn't add implicit cast, and that's why it worked so far (since we've never added implicit `NOT_NULL` cast). However, since the expected type is involved in, I wonder it would be way deeper in the resolution, and instead of `fir2ir`, we may need implicit cast during the resolution?